chapter_06/lib/plan_provider.dart

```diff
@@ -4,13 +4,15 @@ import 'controllers/plan_controller.dart';
 class PlanProvider extends InheritedWidget {
   final _controller = PlanController();
 
-  PlanProvider({Key key, Widget child}) : super(key: key, child: child);
-
-  @override
-  bool updateShouldNotify(InheritedWidget oldWidget) => false;
+  PlanProvider({Key? key, required Widget child})
+      : super(key: key, child: child);
 
   static PlanController of(BuildContext context) {
-    PlanProvider provider = context.dependOnInheritedWidgetOfExactType<PlanProvider>();
+    final provider = context.dependOnInheritedWidgetOfExactType<PlanProvider>()
```

Comment:
If I specify `PlanProvider`, I get an error `A value of type 'PlanProvider?' can't be assigned to a variable of type 'PlanProvider'.`